### ISSUE TITLE
Query GSI 

### ISSUE THREAD

**Issue by sheshankkodam** [circles-learning-labs/ecto_adapters_dynamodb#24]
Is it possible to query with index name? 

For example, I have a user table with index name `user-index` and would like to get all orders by `userId`. Here is the aws cli command 
```
aws dynamodb query --table-name customer --index-name user-index --key-condition-expression "userId = :userId and orderId = :orderId" --expression-attribute-values '{":userId": {"S": "tom"}}'
```

Following is the migration file for User table 

```
defmodule App.Repo.Migrations.AddUserTable do
  use Ecto.Migration

  def change do
    create table(:user, primary_key: false,
             options: [
               global_indexes: [
                 [index_name: "user-index",
                   keys: [:userId, :orderId],
                   provisioned_throughput: [20,20],
                   create_if_not_exists: true]

               ],
               provisioned_throughput: [20,20]
             ]
           ) do
      add :userId, :string, hash_key:    true  # gsi primary key
      add :orderId,  :string, hash_key:    true  # gsi sort key
      timestamps()
    end
  end
end
```

**Comment by sheshankkodam** [comment]
Hello @darrenklein any update on this?

**Comment by darrenklein** [comment]
I believe that any query on those keys will use that index. Off the top of my head - 

```
Repo.all(from m in Model, where: m.userId == "foo" and m.orderId == "bar")
```

@alhambra1 correct me if I'm wrong on this.

**Comment by darrenklein** [comment]
@sheshankkodam Please close this issue if my answer has been sufficient.

**Comment by sheshankkodam** [comment]
yes it worked. Thanks @darrenklein

**Comment by darrenklein** [comment]
@sheshankkodam we've been revisiting this topic, and feel that, although the adapter's default behavior is to select the "best" index based on the query, there are some circumstances where the user may want to be able to specify the index to be used.  Keep an eye on issue #25, it's something we might try to include in the relatively near future.

**Comment by sheshankkodam** [comment]
sure thanks @darrenklein

**Comment by darrenklein** [comment]
@sheshankkodam Ok, this feature is now available in the latest release, version 1.2.0

https://github.com/circles-learning-labs/ecto_adapters_dynamodb/commit/9bcfa89a48e86180d0657dd79286fc8ae8b3e770
